### PR TITLE
Add support for all pre-login tokens

### PR DESCRIPTION
**Pull request by pimeys**
Defined in: https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-tds/60f56408-0188-4cd5-8b90-25c6f2423868
Part of: https://github.com/prisma/prisma/issues/11009

### CODE REVIEW

File: src/tds/codec/pre_login.rs [hunk 1]
```diff
@@ -113,19 +134,69 @@ impl Decode<BytesMut> for PreloginMessage {
...
+                // fed auth
+                0x06 => {
+                    ret.fed_auth_required = cursor.read_u8()? != 0;
+                }
+                // nonce
+                0x07 => {
+                    let mut data = [0u8; 32];
+
+                    for item in data.iter_mut() {
+                        *item = cursor.read_u8()?;
```
Comment (tomhoule) [inline src/tds/codec/pre_login.rs:195]: it looks a bit weird, isn't there a `read()` method to fill the buffer?

File: src/tds/codec/pre_login.rs [hunk 2]
```diff
@@ -113,19 +134,69 @@ impl Decode<BytesMut> for PreloginMessage {
...
+                // instance name
+                0x02 => {
+                    let mut bytes = Vec::new();
+                    let mut next_byte = cursor.read_u8()?;
+
+                    while next_byte != 0x00 {
+                        bytes.push(next_byte);
+                        next_byte = cursor.read_u8()?;
+
+                        continue;
```
Comment (tomhoule) [inline src/tds/codec/pre_login.rs:157]: isn't this continue a no-op?

**Review by tomhoule — APPROVED** [review]
I haven't dived into the protocol, but the code looks good :+1:

File: src/tds/codec/pre_login.rs (hunk 2 above)
Comment (pimeys) [reply src/tds/codec/pre_login.rs:157]: it is.

File: src/tds/codec/pre_login.rs (hunk 1 above)
Comment (pimeys) [reply src/tds/codec/pre_login.rs:195]: true.

**Pull request merged**